PlugIns/DotScene/README.md

```diff
@@ -42,4 +43,46 @@ To add logic properties to the scene you can use the `<userData>` node as follow
         <property data="1.0" name="mass_radius" type="float" />
     </userData>
 </entity>
-```
\ No newline at end of file
+```
+
+## How to use DotScene
+In recent OGRE3D versions DotScene has been incorporated into the main OGRE repo and made into a Plugin.
+So it has to be loaded in `plugins.cfg`, add the following line:
+```
+Plugin=Plugin_DotScene
+```
+Also, it is required to complie against the library (in a similar fashion to OgreBites) located in: `OgreSDK\ogre-1.12.11\lib\OGRE`
+
+Include the header `#include <OgreDotSceneLoader.h>
+`, located in: `OgreSDK\ogre-1.12.11\include\OGRE\Plugins\DotScene`
+
+And to use the library , create a DataStream and pass it to the loader:
+```
+Ogre::String groupName = "Scene";
+Ogre::String filename = "myScene.scene";
+Ogre::SceneNode attachmentNode = mSceneMgr->getRootSceneNode();
+
+Ogre::DataStreamPtr stream(Ogre::Root::openFileStream(filename, groupName));
+
+Ogre::DotSceneLoader *loader = new Ogre::DotSceneLoader();
```

Comment:
instead do
```cpp
Codec::getCodec("scene")->decode(stream, attachmentNode);
```
---
while we are at it, you can add:

if a TerrainGroup is defined in the .scene file, you can get it by:
```cpp
attachmentNode->getUserObjectBindings().getUserAny("TerrainGroup");
```
the type is `std::shared_ptr<Ogre::TerrainGroup>`, hence the `attachmentNode` owns it and will take it down on destruction.

Reply:
That's a strange way to do it...

It is simpler, but does not seem very intuitive that a codec would do that.

Reply:
Are there other UserObjectBindings() registered for the attachmentNode, besides TerrainGroup?

Reply:
All "userData" content is put there for their parent nodes/ entities.

Reply:
> It is simpler, but does not seem very intuitive that a codec would do that.

the Codec is now a central registry for encoding/ decoding data. Previously it was only used for Images, but now it also handles Meshes and Scenes.